tests/python/unittest/test_tvmscript_printer_python_doc_printer.py

```diff
@@ -421,7 +451,590 @@ def test_print_dict_doc(content, expected):
+                ],
+            ),
+            "comment",
+            """
+            # comment
+            while True:
+                x = y
+            """,
+        ),
+        (
+            ForDoc(IdDoc("x"), IdDoc("y"), []),
+            "comment",
+            """
+            # comment
+            for x in y:
+                pass
+            """,
+        ),
+        (
+            ScopeDoc(IdDoc("x"), IdDoc("y"), []),
+            "comment",
+            """
+            # comment
+            with y as x:
+                pass
+            """,
+        ),
+        (
+            ExprStmtDoc(IdDoc("x")),
+            "comment",
+            """
+            x  # comment
+            """,
+        ),
+        (
+            AssertDoc(LiteralDoc(True)),
+            "comment",
+            """
+            assert True  # comment
+            """,
+        ),
+        (
+            ReturnDoc(LiteralDoc(1)),
+            "comment",
+            """
+            return 1  # comment
+            """,
+        ),
+        (
+            get_func_doc_for_class("f"),
+            "comment",
+            '''
+            @wrap
+            def f(x: int, y: int = 1) -> None:
+                """
+                comment
+                """
+                y = x + 1
+                y = y - 1
+            ''',
+        ),
+        (
+            get_func_doc_for_class("f"),
+            "comment line 1\n\ncomment line 3",
+            '''
+            @wrap
+            def f(x: int, y: int = 1) -> None:
+                """
+                comment line 1
+
+                comment line 3
+                """
+                y = x + 1
+                y = y - 1
+            ''',
+        ),
+        (
+            ClassDoc(IdDoc("TestClass"), decorators=[IdDoc("wrap")], body=[]),
+            "comment",
+            '''
+            @wrap
+            class TestClass:
+                """
+                comment
+                """
+                pass
+            ''',
+        ),
+        (
+            ClassDoc(IdDoc("TestClass"), decorators=[IdDoc("wrap")], body=[]),
+            "comment line 1\n\ncomment line 3",
+            '''
+            @wrap
+            class TestClass:
+                """
+                comment line 1
+
+                comment line 3
+                """
+                pass
+            ''',
+        ),
+    ],
+    ids=itertools.count(),
+)
+def test_print_doc_comment(doc, comment, expected):
+    doc.comment = comment
+    assert to_python_script(doc) == format_script(expected)
+
+
+@pytest.mark.parametrize(
+    "doc",
+    [
+        AssignDoc(IdDoc("x"), IdDoc("y"), IdDoc("int")),
+        ExprStmtDoc(IdDoc("x")),
+        AssertDoc(IdDoc("x")),
+        ReturnDoc(IdDoc("x")),
+    ],
+)
+def test_print_invalid_multiline_doc_comment(doc):
+    doc.comment = "1\n2"
+    with pytest.raises(ValueError) as e:
+        to_python_script(doc)
+    assert "cannot have newline" in str(e.value)
```

Comment:
Let's append some epilogues to this file so that it could be runnable with `python /path/to/file`

```python
if __name__ == "__main__":
  tvm.testing.main()
```